crates/services/bls_aggregation/src/bls_agg.rs

```diff
@@ -31,6 +31,65 @@ pub struct AggregatedOperators {
     pub signers_operator_ids_set: HashMap<FixedBytes<32>, bool>,
 }
 
+#[derive(Clone)]
+pub struct TaskMetadata {
+    /// Index of the task
+    task_index: TaskIndex,
+    /// Block the task was created
+    task_created_block: u32, // TODO: Should this be a u64?
+    /// Quorum numbers which should respond to the task
+    quorum_numbers: Vec<u8>,
+    /// Thresholds for each quorum
+    quorum_threshold_percentages: QuorumThresholdPercentages,
+    /// Time before expiry of the task response aggregation
+    time_to_expiry: Duration,
+    // Duration of the window to wait for signatures after quorum is reached
+    window_duration: Duration,
+}
+
+impl TaskMetadata {
+    /// Creates a new instance of the TaskMetadata
```

Comment:
nit: you can enclose symbols with `[ ]` to link them
```suggestion
    /// Creates a new instance of [`TaskMetadata`]
```